diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -159,6 +160,14 @@ fun ASTNode.replaceWhiteSpaceText(beforeNode: ASTNode, text: String) {
 fun ASTNode.getFirstChildWithType(elementType: IElementType): ASTNode? =
         this.findChildByType(elementType)
 
+/**
+ * Checks if a function is anonymous
+ */
+fun ASTNode.isAnonymousFunction() {
```

Comment:
do we have a rule for explicit typing for public function?

Reply:
Missing explicit type can only mean that the function returns `Unit`, so there is no need in such rule. Even in the rule that checks `@return` KDoc tag we allow it to be missing if the function has no explicit return type or explicitly specified `Unit`.

Reply:
But @Cheshiriks wanted to make a boolean function, no?